### PR TITLE
Enough to get mlperf running for 2020.2

### PR DESCRIPTION
**Pull request by G4V**
After a few tweaks, 2020.2 now installs, and `ov_mlperf` compiles and runs successfully on models generated under the 2019 branch.

I would imagine the accuracy will be the same as the older variant, so not greatly useful. May be able to check if the performance has improved?

The building of `ov_mlperf` uses a legacy api that is deprecated and will be removed in 2020.3 so again, of limited use.

Of note is that the call to load the CPU extension in `ov_mlperf` was removed to get it to build. It looks like this is now loaded automagically. This change was identified by comparing the 2019 and 2020 version of the `benchmark_app` sample.

### CODE REVIEW

File: package/lib-openvino/scripts.linux/install.sh [hunk 1]
```diff
@@ -116,15 +116,18 @@ echo "Copying the include files to '${INC_DIR}' ..."
 cp -r ${OPENVINO_DIR}/inference-engine/include/* ${INC_DIR}
 exit_if_error "copying the include files failed"
 
+if [ "${PACKAGE_GIT_CHECKOUT}" == "2019_R3.1" ] || [ "${PACKAGE_GIT_CHECKOUT}" == "2019_R3" ] || [ "${PACKAGE_GIT_CHECKOUT}" == "pre-release" ]; then
```
Comment (psyhtest) [inline package/lib-openvino/scripts.linux/install.sh:119]: Does this mean there's no need for the calibration tool for any 2020 releases?

File: program/mlperf-inference-v0.5/ov_mlperf_cpu/CMakeLists.txt [hunk 2]
```diff
@@ -39,17 +43,22 @@ endif()
 set(OPENVINO_DIR                   "${OPENVINO_DIR}")
 set(OPENVINO_LIB_DIR               "${OPENVINO_DIR}/lib")
 set(OPENVINO_LIBRARY               "${OPENVINO_LIB_DIR}/libinference_engine.so")
-set(OPENVINO_CPU_EXTENSION_LIBRARY "${OPENVINO_LIB_DIR}/libcpu_extension.so")
+if (OPENVINO_VER STREQUAL "2019_R3.1" OR OPENVINO_VER STREQUAL "2019_R3" OR OPENVINO_VER STREQUAL "pre-release")
+  set(OPENVINO_CPU_EXTENSION_LIBRARY "${OPENVINO_LIB_DIR}/libcpu_extension.so")
+elseif(OPENVINO_VER STREQUAL "2020" OR OPENVINO_VER STREQUAL "2020.2")
```
Comment (psyhtest) [inline program/mlperf-inference-v0.5/ov_mlperf_cpu/CMakeLists.txt:48]: How about 2020.1?

Comment (G4V) [reply program/mlperf-inference-v0.5/ov_mlperf_cpu/CMakeLists.txt:48]: Looks like that needs adding.

File: package/lib-openvino/scripts.linux/install.sh (hunk 1 above)
Comment (G4V) [reply package/lib-openvino/scripts.linux/install.sh:119]: The mechanism for calibration has changed. How this is handled will need to be looked into.

**Comment by psyhtest** [comment]
I've managed to build an [Ubuntu 20.04 based image](https://github.com/ctuning/ck-mlperf/blob/master/docker/mlperf-inference-v0.7.openvino/Dockerfile.ubuntu-20.04) with OpenVINO 2020.2. As you say, we will need to look into calibration changes since 2019, but it works with SSD-MobileNet:

```bash
$ export NPROCS=`grep -c processor /proc/cpuinfo` && \
docker run --rm ctuning/mlperf-inference-v0.7.openvino:ubuntu-20.04  \
"ck run ck-openvino:program:mlperf-inference-v0.5 --cmd_key=object-detection --skip_print_timers \
--env.CK_LOADGEN_SCENARIO=Offline --env.CK_LOADGEN_MODE=Accuracy \
--env.CK_LOADGEN_DATASET_SIZE=5000 \
--env.CK_OPENVINO_NTHREADS=$NPROCS --env.CK_OPENVINO_NSTREAMS=$NPROCS --env.CK_OPENVINO_NIREQ=$NPROCS \
&& cat /home/dvdt/CK_REPOS/ck-openvino/program/mlperf-inference-v0.5/tmp/accuracy.txt"

 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.227
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.348
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.246
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.019
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.162
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.523
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.207
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.259
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.259
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.023
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.187
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.597
mAP=22.729%
```

Surprisingly, the accuracy has gone up from`mAP=22.617%` to `mAP=22.729%`.

**Comment by psyhtest** [comment]
Benchmarking on an `g4dn.4xlarge` AWS instance (Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz, 16 virtual cores) shows consistently worse performance with 2020 vs 2019.

### 2019 "pre-release"
```
$ export NPROCS=`grep -c processor /proc/cpuinfo` && docker run --rm ctuning/mlperf-inference-v0.5.openvino:ubuntu-18.04 \
  "ck run ck-openvino:program:mlperf-inference-v0.5 --cmd_key=object-detection --skip_print_timers \
  --env.CK_LOADGEN_SCENARIO=Offline --env.CK_LOADGEN_MODE=Performance --env.CK_LOADGEN_TARGET_QPS=7 \
  --env.CK_OPENVINO_NTHREADS=$NPROCS --env.CK_OPENVINO_NSTREAMS=$NPROCS --env.CK_OPENVINO_NIREQ=$NPROCS \
  && echo '--------------------------------------------------------------------------------' \
  && echo 'mlperf_log_summary.txt' \
  && echo '--------------------------------------------------------------------------------' \
  && cat  /home/dvdt/CK_REPOS/ck-openvino/program/mlperf-inference-v0.5/tmp/mlperf_log_summary.txt \
  && echo '--------------------------------------------------------------------------------'"
...
================================================
MLPerf Results Summary
================================================
SUT name : SUT
Scenario : Offline
Mode     : Performance
Samples per second: 924.659
Result is : INVALID
  Min duration satisfied : NO
  Min queries satisfied : Yes
Recommendations:
 * Increase expected QPS so the loadgen pre-generates a larger (coalesced) query.
```

### 2020.2
```
$ export NPROCS=`grep -c processor /proc/cpuinfo` && docker run --rm ctuning/mlperf-inference-v0.7.openvino:ubuntu-20.04 \
  "ck run ck-openvino:program:mlperf-inference-v0.5 --cmd_key=object-detection --skip_print_timers \
  --env.CK_LOADGEN_SCENARIO=Offline --env.CK_LOADGEN_MODE=Performance --env.CK_LOADGEN_TARGET_QPS=7 \
  --env.CK_OPENVINO_NTHREADS=$NPROCS --env.CK_OPENVINO_NSTREAMS=$NPROCS --env.CK_OPENVINO_NIREQ=$NPROCS \
  && echo '--------------------------------------------------------------------------------' \
  && echo 'mlperf_log_summary.txt' \
  && echo '--------------------------------------------------------------------------------' \
  && cat  /home/dvdt/CK_REPOS/ck-openvino/program/mlperf-inference-v0.5/tmp/mlperf_log_summary.txt \
  && echo '--------------------------------------------------------------------------------'"
...
================================================
MLPerf Results Summary
================================================
SUT name : SUT
Scenario : Offline
Mode     : Performance
Samples per second: 856.027
Result is : INVALID
  Min duration satisfied : NO
  Min queries satisfied : Yes
Recommendations:
 * Increase expected QPS so the loadgen pre-generates a larger (coalesced) query.
```

**Pull request merged**